server/src/main/java/org/elasticsearch/action/admin/indices/template/post/TransportSimulateIndexTemplateAction.java

```diff
@@ -99,30 +98,51 @@ protected SimulateIndexTemplateResponse read(StreamInput in) throws IOException
     @Override
     protected void masterOperation(Task task, SimulateIndexTemplateRequest request, ClusterState state,
                                    ActionListener<SimulateIndexTemplateResponse> listener) throws Exception {
-        ClusterState simulateOnClusterState = state;
+        final ClusterState stateWithTemplate;
         if (request.getIndexTemplateRequest() != null) {
             // we'll "locally" add the template defined by the user in the cluster state (as if it existed in the system)
-            String simulateTemplateToAdd = "simulate_new_template_" + UUIDs.randomBase64UUID().toLowerCase(Locale.ROOT);
-            simulateOnClusterState = indexTemplateService.addIndexTemplateV2(state, request.getIndexTemplateRequest().create(),
+            String simulateTemplateToAdd = "simulate_index_template_" + UUIDs.randomBase64UUID().toLowerCase(Locale.ROOT);
+            // Perform validation for things like typos in component template names
+            MetadataIndexTemplateService.validateV2TemplateRequest(state.metadata(), simulateTemplateToAdd,
+                request.getIndexTemplateRequest().indexTemplate());
+            stateWithTemplate = indexTemplateService.addIndexTemplateV2(state, request.getIndexTemplateRequest().create(),
                 simulateTemplateToAdd, request.getIndexTemplateRequest().indexTemplate());
+        } else {
+            stateWithTemplate = state;
         }
 
-        String matchingTemplate = findV2Template(simulateOnClusterState.metadata(), request.getIndexName(), false);
+        String matchingTemplate = findV2Template(stateWithTemplate.metadata(), request.getIndexName(), false);
         if (matchingTemplate == null) {
             listener.onResponse(new SimulateIndexTemplateResponse(null, null));
             return;
         }
-        Settings settings = resolveSettings(simulateOnClusterState.metadata(), matchingTemplate);
 
-        // empty request mapping as the user can't specify any explicit mappings via the simulate api
-        Map<String, Object> mappings = resolveV2Mappings("{}", simulateOnClusterState, matchingTemplate, xContentRegistry);
-        String mappingsJson = Strings.toString(XContentFactory.jsonBuilder()
-            .startObject()
-            .field(MapperService.SINGLE_MAPPING_NAME, mappings)
-            .endObject());
```

Comment:
@andreidan I had to remove this wrapping because the simulated documents came back with the `_doc` parameter in the mappings, and they can't be sent as new component or index templates with `_doc` in the mappings